docs/resources/workspace_desktop.md

```diff
@@ -0,0 +1,191 @@
+  name = "WorkspaceUserSecurityGroup"
+}
+
+resource "huaweicloud_workspace_desktop" "test" {
+  flavor_id  = var.flavor_id
+  image_type = "market"
+  image_id   = var.image_id
+
+  availability_zone = data.huaweicloud_availability_zones.test.names[0]
+  vpc_id            = var.vpc_id
+  security_groups   = setunion(data.huaweicloud_networking_secgroups.test.security_groups[*].id,
+    [var.security_group_id])
+
+  nics {
+    network_id = var.network_id
+  }
+
+  name       = var.desktop_name
+  user_name  = "TestUser"
+  user_email = "terraform@example.com"
+  user_group = "administrators"
+
+  root_volume {
+    type = "SAS"
+    size = 80
+  }
+
+  data_volume {
+    type = "SAS"
+    size = 50
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) The region in which to create the Workspace desktop resource.
+  If omitted, the provider-level region will be used. Changing this will create a new resource.
+
+* `flavor_id` - (Required, String) Specifies the flavor ID of desktop.
+
+* `image_type` - (Required, String, ForceNew) Specifies the image type. The valid values are as follows:
+  + **market**: The market image.
+  + **gold**: The public image.
+  + **private**: The private image.
+
+  Changing this will create a new resource.
+
+* `image_id` - (Required, String, ForceNew) Specifies the image ID to create the desktop.
+  Changing this will create a new resource.
+
+* `vpc_id` - (Required, String, ForceNew) Specifies the VPC ID to which the desktop belongs.
+  Changing this will create a new resource.
+
+* `user_name` - (Required, String, ForceNew) Specifies the user name to which the desktop belongs.
+  The name can contain `1` to `20` characters, only letters, digits, hyphens (-) and underscore (_) are allowed.
+  The name must start with a letter. Changing this will create a new resource.
+
+* `user_email` - (Required, String, ForceNew) Specifies the user email.
+  Some operations on the desktop (such as creation, deletion) will notify the user by sending an email.
+  Changing this will create a new resource.
+
+* `root_volume` - (Required, List) Specifies the configuration of system volume.
+  The [object](#desktop_volume) structure is documented below.
+
+* `data_volume` - (Optional, List) Specifies the configuration of data volumes.
+  The [object](#desktop_volume) structure is documented below.
+
+* `availability_zone` - (Optional, String, ForceNew) Specifies the availability zone where the desktop is located.
+  Changing this will create a new resource.
+
+* `security_groups` - (Optional, String, ForceNew) Specifies the ID list of security groups.
+  In addition to the custom security group, it must also contain a security group called **WorkspaceUserSecurityGroup**.
+  Changing this will create a new resource.
+
+* `user_group` - (Required), String, ForceNew) Specifies the user group to which the desktop belongs.
```

Comment:
extra `)`, and how about move this to line 89

Reply:
fixed.